lib/compat/wordpress-6.4/fonts/font-face/class-wp-font-face-resolver.php

```diff
@@ -76,7 +76,15 @@ private static function parse_settings( array $settings ) {
 						$fonts[ $font_family_name ] = array();
 					}
 
-					$fonts[ $font_family_name ] = static::convert_font_face_properties( $definition['fontFace'], $font_family_name );
+					$font_faces_to_add = static::convert_font_face_properties( $definition['fontFace'], $font_family_name );
+					foreach ( $font_faces_to_add as $font_face_to_add ) {
+						$offset = static::search_font_face( $font_face_to_add, $fonts[ $font_family_name ] );
+						if ( false !== $offset ) {
+							$fonts[ $font_family_name ][ $offset ] = array_merge( $fonts[ $font_family_name ][ $offset ], $font_face_to_add );
```

Comment:
The use of array_merge here combines the properties of the existing font face (`$fonts[ $font_family_name ][ $offset ]`) with those of the new font face (`$font_face_to_add`). This operation prioritizes the properties from `$font_face_to_add` when the same keys exist in both arrays. For instance, if both arrays contain a `src` property, the `src` from `$font_face_to_add` might overwrite the one in the existing font face.

Should we keep the original font face when all `font-family` & `font-style` & `font-weight` are the same?

I thought just `$fonts[ $font_family_name ][] = $font_face_to_add;` does a good job of merging the original and the installed font with **different** variants. 
WDYT? 💭 @arthur791004

Reply:
That's a fair point. IMHO, I assume it's fine to override the src if all the `font-family` & `font-style` & `font-weight` are the same as the fonts here are for generating and printing font-face styles to the HTML. I don't think we have to print the same font face multiple times. Do you have any concerns about that?

Reply:
I was somewhat hesitant to overwrite the built-in fonts because they might have already been set. However, as you mentioned, 

>I assume it's fine to override the src if all the font-family & font-style & font-weight are the same as the fonts here are for generating and printing font-face styles to the HTML

it should indeed be fine. 👍